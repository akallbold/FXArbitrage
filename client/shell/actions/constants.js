/* Organizattions */

export const SET_CURRENT_USER = 'user.SET_CURRENT_USER'
export const SET_CURRENT_USER_INITIAL = 'organization.SET_CURRENT_USER_INITIAL'
export const UPDATED_CURRENT_USER = 'organization.UPDATED_USER_ORGANIZATION'
