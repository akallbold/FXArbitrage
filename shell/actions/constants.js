/* Organizattions */

export const SET_CURRENT_ORGANIZATION = 'organization.SET_CURRENT_ORGANIZATION'
export const SET_CURRENT_ORGANIZATION_INITIAL = 'organization.SET_CURRENT_ORGANIZATION_INITIAL'
export const UPDATED_CURRENT_ORGANIZATION = 'organization.UPDATED_CURRENT_ORGANIZATION'

/* Side Nav */
export const TOGGLE_SIDE_NAV = 'sideNav.TOGGLE_SIDE_NAV'

/* Account Menu */
export const VIEW_PROFILE = 'accountMenu.VIEW_PROFILE'
export const SIGN_OUT = 'accountMenu.SIGN_OUT'
export const REDIRECT_TO_HELP = 'accountMenu.REDIRECT_TO_HELP'
export const HIDE_ACCOUNT_MENU = 'accountMenu.HIDE_ACCOUNT_MENU'
export const SHOW_ACCOUNT_MENU = 'accountMenu.SHOW_ACCOUNT_MENU'

/* Platforms */
export const SET_ACTIVE_PLATFORMS = 'channel.SET_ACTIVE_PLATFORMS'
export const SET_CURRENT_PLATFORM = 'channel.SET_CURRENT_PLATFORM'
export const SET_CURRENT_PLATFORM_INITIAL = 'channel.SET_CURRENT_PLATFORM_INITIAL'
