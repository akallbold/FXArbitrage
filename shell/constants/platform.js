export const FACEBOOK_PLATFORM_NAME = 'facebook'
export const TWITTER_PLATFORM_NAME = 'twitter'
export const SNAPCHAT_PLATFORM_NAME = 'snapchat'
export const PINTEREST_PLATFORM_NAME = 'pinterest'
export const PLATFORM_NAMES = {
  FACEBOOK: FACEBOOK_PLATFORM_NAME,
  TWITTER: TWITTER_PLATFORM_NAME,
  SNAPCHAT: SNAPCHAT_PLATFORM_NAME,
  PINTEREST: PINTEREST_PLATFORM_NAME
}
