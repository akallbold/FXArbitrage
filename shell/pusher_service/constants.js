export const PUSHER_NOTIFICATION_RECEIVED = 'PUSHER_NOTIFICATION_RECEIVED'
export const PUSHER_DATA_RECEIVED = 'PUSHER_DATA_RECEIVED'
export const CLEAR_PUSHER_DATA = 'CLEAR_PUSHER_DATA'
export const PUSHER_CONNECTION_ESTABLISHED = 'PUSHER_CONNECTION_ESTABLISHED'
export const EVENT_NEW_NOTIFICATION = 'new_notification'
