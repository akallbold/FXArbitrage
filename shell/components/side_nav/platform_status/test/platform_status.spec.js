import React from 'react'
import PropTypes from 'prop-types'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'
import { PLATFORM_NAMES } from 'constants/platform'
import {
  PlatformStatus,
  PlatformStatusItem,
  platformLinkData,
  platformClickHandler
} from 'components/side_nav/platform_status/platform_status'

describe('<PlatformStatus />', () => {
  const mockStore = configureMockStore()
  const store = mockStore({ router: {} })
  const platformStatuses = {
    currentPlatform: 'snapchat',
    facebook: false,
    snapchat: true,
    twitter: false,
    pinterest: true,
    error: null,
    loading: false
  }

  const options = {
    context: { store },
    childContextTypes: { store: PropTypes.object.isRequired }
  }

  const platformIdsOrder = [
    PLATFORM_NAMES.FACEBOOK,
    PLATFORM_NAMES.TWITTER,
    PLATFORM_NAMES.PINTEREST,
    PLATFORM_NAMES.SNAPCHAT
  ]

  it('should have platforms in correct order', () => {
    const platformIds = platformLinkData.map(item => item.id)
    expect(platformIds).toEqual(platformIdsOrder)
  })

  it('should render properly given props', () => {
    const component = mount(
      <PlatformStatus
        platformStatuses={platformStatuses}
        currentBrandId={'5'}
        selectedChannel={'pinterest'}
      />,
      options
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('<PlatformStatusItem />', () => {
  it('should render properly given props', () => {
    const itemData = platformLinkData[0]
    const component = shallow(
      <PlatformStatusItem {...itemData} status />
    )

    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('platformClickHandler', () => {
  describe('for same channel', () => {
    it('should return a preventDefault function click handler', () => {
      const clickHandler = platformClickHandler({ selectedChannel: 'pinterest', channelName: 'pinterest' })
      expect(clickHandler).toBeInstanceOf(Function)
    })
  })

  describe('for different channel', () => {
    it('should return a null click handler', () => {
      const clickHandler = platformClickHandler({ selectedChannel: 'pinterest', channelName: 'facebook' })
      expect(clickHandler).toEqual(null)
    })
  })
})
