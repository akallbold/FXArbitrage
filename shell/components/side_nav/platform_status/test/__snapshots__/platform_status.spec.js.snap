// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformStatus /> should render properly given props 1`] = `
<PlatformStatus
  currentBrandId="5"
  platformStatuses={
    Object {
      "currentPlatform": "snapchat",
      "error": null,
      "facebook": false,
      "loading": false,
      "pinterest": true,
      "snapchat": true,
      "twitter": false,
    }
  }
  selectedChannel="pinterest"
>
  <div>
    <div
      className="heading"
    >
      Platforms
    </div>
    <PlatformStatusItem
      current={false}
      handlePlatformClick={null}
      icon={<test-file-stub />}
      id="facebook"
      label="Facebook"
      status={false}
      url="/brands/5/facebook/manager/campaigns"
    >
      <Connect(Link)
        className="item false"
        href="/brands/5/facebook/manager/campaigns"
        onClick={null}
      >
        <Link
          className="item false"
          href="/brands/5/facebook/manager/campaigns"
          location={Object {}}
          onClick={null}
          push={[Function]}
          replace={[Function]}
        >
          <a
            className="item false"
            href="/brands/5/facebook/manager/campaigns"
            onClick={[Function]}
          >
            <div
              className="itemContent"
            >
              <span
                className="icon"
              >
                <test-file-stub />
              </span>
              <span
                className="name"
                data-field="name"
              >
                Facebook
              </span>
            </div>
          </a>
        </Link>
      </Connect(Link)>
    </PlatformStatusItem>
    <PlatformStatusItem
      current={false}
      handlePlatformClick={null}
      icon={<test-file-stub />}
      id="twitter"
      label="Twitter"
      status={false}
      url="/brands/5/twitter/manager/campaigns"
    >
      <Connect(Link)
        className="item false"
        href="/brands/5/twitter/manager/campaigns"
        onClick={null}
      >
        <Link
          className="item false"
          href="/brands/5/twitter/manager/campaigns"
          location={Object {}}
          onClick={null}
          push={[Function]}
          replace={[Function]}
        >
          <a
            className="item false"
            href="/brands/5/twitter/manager/campaigns"
            onClick={[Function]}
          >
            <div
              className="itemContent"
            >
              <span
                className="icon"
              >
                <test-file-stub />
              </span>
              <span
                className="name"
                data-field="name"
              >
                Twitter
              </span>
            </div>
          </a>
        </Link>
      </Connect(Link)>
    </PlatformStatusItem>
    <PlatformStatusItem
      current={true}
      handlePlatformClick={[Function]}
      icon={<test-file-stub />}
      id="pinterest"
      label="Pinterest"
      status={true}
      url="/brands/5/pinterest/manager/campaigns"
    >
      <Connect(Link)
        className="item itemActive"
        href="/brands/5/pinterest/manager/campaigns"
        onClick={[Function]}
      >
        <Link
          className="item itemActive"
          href="/brands/5/pinterest/manager/campaigns"
          location={Object {}}
          onClick={[Function]}
          push={[Function]}
          replace={[Function]}
        >
          <a
            className="item itemActive"
            href="/brands/5/pinterest/manager/campaigns"
            onClick={[Function]}
          >
            <div
              className="itemContent"
            >
              <span
                className="icon"
              >
                <test-file-stub />
              </span>
              <span
                className="name"
                data-field="name"
              >
                Pinterest
              </span>
              <span
                className="status"
              />
            </div>
          </a>
        </Link>
      </Connect(Link)>
    </PlatformStatusItem>
    <PlatformStatusItem
      current={false}
      handlePlatformClick={null}
      icon={<test-file-stub />}
      id="snapchat"
      label="Snapchat"
      status={true}
      url="/brands/5/snapchat/manager/campaigns"
    >
      <Connect(Link)
        className="item false"
        href="/brands/5/snapchat/manager/campaigns"
        onClick={null}
      >
        <Link
          className="item false"
          href="/brands/5/snapchat/manager/campaigns"
          location={Object {}}
          onClick={null}
          push={[Function]}
          replace={[Function]}
        >
          <a
            className="item false"
            href="/brands/5/snapchat/manager/campaigns"
            onClick={[Function]}
          >
            <div
              className="itemContent"
            >
              <span
                className="icon"
              >
                <test-file-stub />
              </span>
              <span
                className="name"
                data-field="name"
              >
                Snapchat
              </span>
              <span
                className="status"
              />
            </div>
          </a>
        </Link>
      </Connect(Link)>
    </PlatformStatusItem>
  </div>
</PlatformStatus>
`;

exports[`<PlatformStatusItem /> should render properly given props 1`] = `
<Connect(Link)
  className="item undefined"
  href="/facebook"
>
  <div
    className="itemContent"
  >
    <span
      className="icon"
    >
      <test-file-stub />
    </span>
    <span
      className="name"
      data-field="name"
    >
      Facebook
    </span>
    <span
      className="status"
    />
  </div>
</Connect(Link)>
`;
